Feature/split arch

kristiankielhofner opened the pull request.

kristiankielhofner committed:
Initial support for coqui TTS, disable SpeechT5 in WIS by default

kristiankielhofner committed:
WIP: Remove SpeechT5 support

kristiankielhofner committed:
Bye bye SpeechT5 - it wasn't great while it lasted

kristiankielhofner committed:
Use only the first GPU by default

kristiankielhofner committed:
Update requirements

kristiankielhofner committed:
Remove chatbot support

kristiankielhofner committed:
Dynamic library fixes

kristiankielhofner committed:
GPU batch log tweak

kristiankielhofner committed:
Initial flake8 fixes

kristiankielhofner committed:
Nvidia driver version check

kristiankielhofner committed:
Fix CPU handling and Nvidia detection

kristiankielhofner committed:
Batch works on CPU and GPU, remove GPU from log line to not confuse people

kristiankielhofner committed:
Supress warning from coqui on generation

kristiankielhofner committed:
Add support for X-API-Key authentication with Willow and other clients

kristiankielhofner committed:
Generate x25519 key by default

kristiankielhofner committed:
Implement WIS Basic HTTP Auth

Add users like so (using bcrypt):

./utils.sh htpasswd -B /nginx/htpasswd my_user

kristiankielhofner committed:
Support configuring of nginx tag, fix coqui tag config, bump nginx to 1.25.3 by default

kristiankielhofner committed:
Disable 3DES cipher suites in nginx - it's 2023

kristiankielhofner committed:
Generate 2048 DH params

kristiankielhofner committed:
Wrap openssl calls to nginx docker images

This ensures we don't use potentially insecure openssl from host

Additionally, it sets proper permissions for key files on host

kristiankielhofner committed:
Add auth-basic.config.template

kristiankielhofner committed:
Mutiple security fixes:

- Build our own nginx container to include htpasswd so permissions are set
- Wrap all openssl and related commands to WIS nginx docker container

kristiankielhofner committed:
Massive CPU performance improvement.

Before:
3067.8740000000003 small/2

After:
2094.071 small/2

- Build with clang 16
- Bump oneDNN (with faster int8)
- Bump oneAPI

kristiankielhofner committed:
Based on extensive profiling:

- TensorRT 23.05
- oneAPI 2023.1.0
- oneDNN 3.1.1
- clang 14

Also include ctranslate2 fork with whisper-distil support and transformers 4.35.0

kristiankielhofner committed:
Add useradd/del/list helper args

kristiankielhofner committed:
Add little help text for useradd

kristiankielhofner committed:
Add header Cache-Control: public so CF will cache TTS even with auth

kristiankielhofner committed:
Add header Cache-Control: public so CF will cache TTS even with auth

kristiankielhofner committed:
Initial pass at ensuring proper permissions on the nginx cache

kristiankielhofner committed:
Update to official ctranslate2 release for distil-whisper and large-v3

kristiankielhofner committed:
Add detect compute to relevant args

kristiankielhofner committed:
Add support for Coqui XTTS

kristiankielhofner committed:
Incorporate some XTTS tweaks from their HF space

kristiankielhofner committed:
Check supported language, warm TTS on start

kristiankielhofner committed:
For consistency update docker-compose-cpu - XTTS not supported anyway

kristiankielhofner committed:
XTTS: Support passing of all xtts model args

kristiankielhofner committed:
Update to latest from xtts-streaming-server
Pin FROM docker container
Set enable_text_splitting = True
Add hu and ko, change chinese to zh
Limit range of top_p based on model specs
Update comments

kristiankielhofner committed:
Add URL param nocache to /api/tts to disable TTS caching
Set to something: ?nocache=true

kristiankielhofner committed:
Support adding custom speakers with POST to /api/tts

curl -F 'audio_file=@yourwav.wav' \
  'https://wis:19000/api/tts?speaker=your_speaker_name'

Then you can immediately pass speaker=your_speaker_name to GET

kristiankielhofner committed:
Update XTTS Dockerfile to copy all speaker json in dir

kristiankielhofner committed:
Add female and male speakers
Rename speecht5 to CLB

kristiankielhofner committed:
Support CPU - SLOW

kristiankielhofner committed:
Bump Coqui

kristiankielhofner committed:
Update to latest XTTS streaming-server-base with model DL fix

kristiankielhofner committed:
Update XTTS base image

kristiankielhofner committed:
Turn down default XTTS temperature to 0.1

kristiankielhofner committed:
Revert "Update to official ctranslate2 release for distil-whisper and large-v3"

This reverts commit 3502dc3285d13adb37c40cc0a6fe942cea1d0824.

kristiankielhofner committed:
Revert "Based on extensive profiling:"

This reverts commit ea57276309c2ecfda741615c731740af03b35291.

kristiankielhofner committed:
Revert "Massive CPU performance improvement."

This reverts commit 258c76aedad7b8f0c5642e242bf8a99cf8078e42.

kristiankielhofner committed:
Revert "WIP: Initial CUDA 12 support"

This reverts commit 237188b41f0ae374231b96a6b39b4041102c6838.

kristiankielhofner committed:
Update all deps

ctranslate2 now natively supports CUDA 12

Warning: this breaks custom_settings.py due to pydantic changes

kristiankielhofner committed:
Update nginx to latest

kristiankielhofner committed:
Update coqui to latest

kristiankielhofner committed:
Update xtts to latest

stintel committed:
CI: replace flake8 check with black

stintel committed:
treewide: format Python code with black

stintel commented:
This adds a new container for coqui TTS, with various fixes.
Chatbot support is removed, as this is not a focus area for Willow.
Python flake8 check is replaced with black.

stintel committed:
Dockerfile: install pkg-congig

Fixes the following build error:

Collecting av==11.0.0 (from -r requirements.txt (line 12))
  Downloading av-11.0.0.tar.gz (3.7 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.7/3.7 MB 20.8 MB/s eta 0:00:00
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'error'
  error: subprocess-exited-with-error

  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [1 lines of output]
      pkg-config is required for building PyAV
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

stintel committed:
Dockerfile: order apt package alphabetically

And put each on a separate line. Improves readability.

stintel committed:
Dockerfile: install ffmpeg build dependencies

Fixes the following build error:

Collecting av==11.0.0 (from -r requirements.txt (line 12))
  Using cached av-11.0.0.tar.gz (3.7 MB)
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'error'
  error: subprocess-exited-with-error

  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [29 lines of output]
      Package libavformat was not found in the pkg-config search path.
      Perhaps you should add the directory containing `libavformat.pc'
      to the PKG_CONFIG_PATH environment variable
      No package 'libavformat' found
      Package libavcodec was not found in the pkg-config search path.
      Perhaps you should add the directory containing `libavcodec.pc'
      to the PKG_CONFIG_PATH environment variable
      No package 'libavcodec' found
      Package libavdevice was not found in the pkg-config search path.
      Perhaps you should add the directory containing `libavdevice.pc'
      to the PKG_CONFIG_PATH environment variable
      No package 'libavdevice' found
      Package libavutil was not found in the pkg-config search path.
      Perhaps you should add the directory containing `libavutil.pc'
      to the PKG_CONFIG_PATH environment variable
      No package 'libavutil' found
      Package libavfilter was not found in the pkg-config search path.
      Perhaps you should add the directory containing `libavfilter.pc'
      to the PKG_CONFIG_PATH environment variable
      No package 'libavfilter' found
      Package libswscale was not found in the pkg-config search path.
      Perhaps you should add the directory containing `libswscale.pc'
      to the PKG_CONFIG_PATH environment variable
      No package 'libswscale' found
      Package libswresample was not found in the pkg-config search path.
      Perhaps you should add the directory containing `libswresample.pc'
      to the PKG_CONFIG_PATH environment variable
      No package 'libswresample' found
      pkg-config could not find libraries ['avformat', 'avcodec', 'avdevice', 'avutil', 'avfilter', 'swscale', 'swresample']
      [end of output]

stintel committed:
Dockerfile: drop --ipc=host

Fixes the following error with podman:

Error: invalid config provided: cannot set shmsize when running in the {host } IPC Namespace

stintel committed:
utils.sh: ensure cache directory exists

stintel committed:
Revert "utils.sh: listen on IPv4 and IPv6 by default"

I've never tested this with Docker compose and it seems to be causing a
lot of issues. No IPv6 then for now.

This reverts commit eb27c559e1fd7955e92d9da4882b8b87d3deb9fc.

stintel committed:
README: drop LLM/chat references

We no longer support that. People can run ollama next to WIS.

stintel commented:
Build-tested with Docker and Podman. Runtime tested on RTX A4000. Time to finally merge this.
